client/src/components/modal-forms/HireForm.js

```diff
@@ -17,66 +20,115 @@ const hireInitialValues = {
   acknowledge: false,
 };
 
-export const HireForm = ({ initialValues, validationSchema, onSubmit, onClose, sites }) => {
+export const HireForm = ({ onSubmit, onClose, sites }) => {
+  const [phases, setPhases] = useState([]);
+
+  const handleCurrentPhase = async (siteId) => {
+    const site = sites.filter((site) => site.siteId === siteId)[0];
+    let phases = await fetchSitePhases(site.id);
+    setPhases(phases);
+  };
+
   return (
     <Formik initialValues={hireInitialValues} validationSchema={HireFormSchema} onSubmit={onSubmit}>
-      {({ submitForm, values }) => (
-        <FormikForm>
-          <Box>
-            <Field
-              name='nonHcapOpportunity'
-              component={RenderCheckbox}
-              label='Non-HCAP Opportunity'
-            />
-            {values.nonHcapOpportunity && (
-              <>
-                <Field name='positionTitle' component={RenderTextField} label='* Position Title' />
-                <Field
-                  name='positionType'
-                  component={RenderSelectField}
-                  label='* Position Type'
-                  options={[
-                    { value: 'Full-Time', label: 'Full-Time' },
-                    { value: 'Part-Time', label: 'Part-Time' },
-                    { value: 'Casual', label: 'Casual' },
-                  ]}
-                />
-              </>
-            )}
-            <Field
-              name='hiredDate'
-              component={RenderDateField}
-              maxDate={getTodayDate()}
-              label='* Date Hired'
-            />
-            <Field name='startDate' component={RenderDateField} label='* Start Date' />
-            <Field
-              name='site'
-              component={RenderSelectField}
-              label='* Site'
-              options={sites.map((siteDetail) => ({
-                value: siteDetail.siteId,
-                label: siteDetail.siteName,
-              }))}
-            />
-            <Field
-              name='acknowledge'
-              component={RenderCheckbox}
-              label='I acknowledge that the participant has accepted the offer in writing.'
-            />
-          </Box>
-          <Box mt={3}>
-            <Grid container spacing={2} justify='flex-end'>
-              <Grid item>
-                <Button onClick={onClose} color='default' text='Cancel' />
-              </Grid>
-              <Grid item>
-                <Button onClick={submitForm} variant='contained' color='primary' text='Submit' />
+      {({ submitForm, values, setFieldValue }) => {
+        const currentPhase =
+          values.hiredDate && values.site
+            ? phases?.filter(
+                (phase) =>
+                  Date.parse(phase.startDate) <= Date.parse(values.hiredDate) &&
+                  Date.parse(values.hiredDate) <= Date.parse(phase.endDate)
+              )[0]
+            : null;
+
+        return (
```

Comment:
For the sake of clarity I would prefer to see some of this logic brought out into a function prior to `return`. Not a huge fan of defining a function that returns JSX *inside the return JSX of a function*.